includes/blocks/class-sensei-course-outline-block.php

```diff
@@ -238,4 +236,82 @@ public function render_course_outline_block( $attributes ) {
 
 	}
 
+	/**
+	 * Initialize notices.
+	 *
+	 * @internal
+	 *
+	 * @since $$next-version$$
+	 */
+	public function frontend_notices() {
+		$post = get_post();
+
+		if ( ! $post || 'course' !== $post->post_type || is_admin() ) {
+			return;
+		}
+
+		$course_id        = $post->ID;
+		$structure        = Sensei_Course_Structure::instance( $course_id )->get( 'view' );
+		$has_draft        = $this->has_draft( $structure );
+		$edit_cta_message = '';
+
+		// Complete message with CTA if user can edit the course.
+		if ( Sensei_Course::can_current_user_edit_course( $course_id ) ) {
```

Comment:
I added the CTA in a separate logic because draft lessons will be displayed based on the capability: `read_private_posts`. See: https://github.com/Automattic/sensei/blob/d53e6a4f1533cf5166fae99bafd4b012dd5f91cf/includes/class-sensei-course-structure.php#L81

So it doesn't mean that a user seeing that will be able to edit the course.